typescript/src/entities/products.ts

```diff
@@ -28,4 +29,17 @@ export class Product extends Entity {
     this._name = newName;
     this._isDirty = true;
   }
+
+  @Product.property("categories", "Category")
```

Comment:
why do i have to pass a string here instead of the actual `Category` class? because, in the case of recursive imports like here, one or both of the classes is undefined at run time, because the class object is not constructed yet during the import phase.

all entities will not be totally built until the imports have all finished, which is too late for us to be able to use it here. so, the string defers until later in runtime, and goes and looks up the entity by name on the `Merchi` class.

Reply:
:+1: it is a classic problem in orm and python sdk

I think it is the best option we can have so far.